project-management-api/src/main/java/com/xwiki/projectmanagement/internal/OpenProjectClient.java

```diff
@@ -0,0 +1,135 @@
+
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.slf4j.Logger;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.livedata.LiveDataQuery;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.xwiki.projectmanagement.ProjectManagementClient;
+import com.xwiki.projectmanagement.dto.OpenProjectWorkItem;
+import com.xwiki.projectmanagement.exception.WorkItemCreationException;
+import com.xwiki.projectmanagement.exception.WorkItemDeletionException;
+import com.xwiki.projectmanagement.exception.WorkItemNotFoundException;
+import com.xwiki.projectmanagement.exception.WorkItemRetrievalException;
+import com.xwiki.projectmanagement.exception.WorkItemUpdatingException;
+import com.xwiki.projectmanagement.model.PaginatedResult;
+import com.xwiki.projectmanagement.model.WorkItem;
+
+/**
+ * Open project client.
+ *
+ * @version $Id$
+ */
+@Component
+@Named("open-project-client")
+@Singleton
+public class OpenProjectClient implements ProjectManagementClient
+{
+    @Inject
+    private Logger logger;
+
+    @Override
+    public WorkItem getWorkItem(String workItemId) throws WorkItemNotFoundException
+    {
+        return null;
+    }
+
+    @Override
+    public PaginatedResult<WorkItem> getWorkItems(int page, int pageSize, List<LiveDataQuery.Filter> filters)
+        throws WorkItemRetrievalException
+    {
+        PaginatedResult<WorkItem> paginatedResult = new PaginatedResult<>();
+
+        List<WorkItem> workItems = new ArrayList<>();
+
+        try {
+            HttpRequest request = HttpRequest
+                .newBuilder()
+                .header("Accept", "application/json")
+                .header("Authorization", "Bearer " + "")
+                .uri(new URI("http://localhost:8080/api/v3/work_packages"))
+                .GET()
+                .build();
+
+            HttpClient client = HttpClient.newHttpClient();
+            HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
+            String body = response.body();
+
+            ObjectMapper objectMapper = new ObjectMapper();
+            objectMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+            JsonNode rootNode = objectMapper.readTree(body);
+
+            JsonNode embeddedNode = rootNode.path("_embedded");
+
+            JsonNode elementsNode = embeddedNode.path("elements");
+
+            List<OpenProjectWorkItem> workItemList = objectMapper
+                .convertValue(elementsNode, new TypeReference<List<OpenProjectWorkItem>>()
```

Comment:
I dont really see the advantage of using the additional OpenProjectWorkItem class. We get the http response that we deserialze in a json, we then transform it into the dto and then we convert it into the work item model. We might as well cut the middle man and transform the deserialized json into the final model. We can separate the covnersion logic into some other class that specializes with that.

Let me know if there is some advantage that the dto offers and that i cannot see